wardrobe/timestamp/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+        );
+
+        // Make sure the block height from the previous timestamp matches the previous block height.
+        ensure!(
+            new_timestamp.block == old_timestamp.block + 1,
+            Self::Error::PreviousTimestampWrongHeight,
+        );
+
+        Ok(0)
+    }
+
+    fn is_inherent(&self) -> bool {
+        true
+    }
+}
+
+impl<V: Verifier + From<UpForGrabs>, T: TimestampConfig + 'static> TuxedoInherent<V, Self>
+    for SetTimestamp<T>
+{
+    type Error = sp_timestamp::InherentError;
+    const INHERENT_IDENTIFIER: sp_inherents::InherentIdentifier = sp_timestamp::INHERENT_IDENTIFIER;
+
+    fn create_inherent(
+        authoring_inherent_data: &InherentData,
+        previous_inherent: Option<(Transaction<V, Self>, H256)>,
+    ) -> tuxedo_core::types::Transaction<V, Self> {
+        // Extract the current timestamp from the inherent data
+        let timestamp_millis: u64 = authoring_inherent_data
+            .get_data(&sp_timestamp::INHERENT_IDENTIFIER)
+            .expect("Inherent data should decode properly")
+            .expect("Timestamp inherent data should be present.");
+        let new_timestamp = Timestamp {
+            time: timestamp_millis,
+            block: T::block_height(),
+        };
+
+        log::debug!(
+            target: LOG_TARGET,
+            "🕰️🖴 Local timestamp while creating inherent i:: {timestamp_millis}"
+        );
+
+        let mut peeks = Vec::new();
+        match (previous_inherent, T::block_height()) {
+            (None, 1) => {
+                // This is the first block hack case.
+                // We don't need any inputs, so just do nothing.
+            }
+            (None, _) => panic!("Attemping to construct timestamp inherent with no previous inherent (and not block 1)."),
+            (Some((_previous_inherent, previous_id)), _) => {
+                // This is the the normal case. We create a full previous to peek at.
+
+                // We are given the entire previous inherent in case we need data from it or need to scrape the outputs.
+                // But out transactions are simple enough that we know we just need the one and only output.
+                peeks.push(OutputRef {
+                    tx_hash: previous_id,
+                    // There is always 1 output, so we know right where to find it.
+                    index: 0,
+                });
+            }
+        }
+
+        let new_output = Output {
+            payload: new_timestamp.into(),
+            verifier: UpForGrabs.into(),
+        };
+
+        Transaction {
+            inputs: Vec::new(),
+            peeks,
+            outputs: vec![new_output],
+            checker: Self::default(),
+        }
+    }
+
+    fn check_inherent(
+        importing_inherent_data: &InherentData,
+        inherent: Transaction<V, Self>,
+        result: &mut CheckInherentsResult,
+    ) {
+        // Extract the local view of time from the inherent data
```

Comment:
This comment is obvious I would remove it

Reply:
fixed in 6409334ff59118cda4dc786271e5e6ece396bd23